src/types/changeset.rs

```diff
@@ -0,0 +1,76 @@
+use std::ops::Deref;
+
+use bdk_core::Merge;
+use bdk_wallet::{
+    serde_json::{from_str, to_string},
+    ChangeSet as BdkChangeSet,
+};
+use serde_wasm_bindgen::{from_value, to_value};
+use wasm_bindgen::{prelude::wasm_bindgen, JsValue};
+
+use crate::result::JsResult;
+
+/// A changeset for [`Wallet`].
+#[wasm_bindgen]
+#[derive(Debug, PartialEq)]
+pub struct ChangeSet {
+    changeset: BdkChangeSet,
+}
+
+#[wasm_bindgen]
+impl ChangeSet {
+    /// Merge another [`ChangeSet`] into itself.
+    pub fn merge(&mut self, other: ChangeSet) {
+        self.changeset.merge(other.into());
+    }
+
+    pub fn is_empty(&self) -> bool {
+        self.changeset.is_empty()
+    }
+
+    /// Serialize `ChangeSet` to JSON.
+    pub fn to_json(&self) -> String {
+        to_string(&self.changeset).expect("Serialization should not fail")
+    }
```

Comment:
Is there a more robust way to handle the error instead of just using `expect`? Or is the idea to just let it terminate? I see that `to_json` and `to_js` are handling errors differently

Reply:
This is an assertion and not an exception, that's why you have an `expect`. This cannot fail as the `ChangeSet` type is Serializable so it can only fail if the `bdk` package itself has a bug where for some reason the ChangeSet type would fail to serialize to JSON and that is not caught in their tests, it will be caught in our tests though.